Build fails with python3.14

From https://bugzilla.redhat.com/show_bug.cgi?id=2416748#c3:
```
[124/281] /usr/bin/g++ -DHAVE_STAT -DHAVE_STAT64 -DSiloPy_EXPORTS -D_LARGEFILE64_SOURCE -I/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/src/silo -I/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/redhat-linux-build/include -isystem /usr/include/python3.14 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -fPIC -MD -MT tools/python/CMakeFiles/SiloPy.dir/pysilo.cpp.o -MF tools/python/CMakeFiles/SiloPy.dir/pysilo.cpp.o.d -o tools/python/CMakeFiles/SiloPy.dir/pysilo.cpp.o -c /builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.cpp
FAILED: [code=1] tools/python/CMakeFiles/SiloPy.dir/pysilo.cpp.o 
/usr/bin/g++ -DHAVE_STAT -DHAVE_STAT64 -DSiloPy_EXPORTS -D_LARGEFILE64_SOURCE -I/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/src/silo -I/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/redhat-linux-build/include -isystem /usr/include/python3.14 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -mtls-dialect=gnu2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -fPIC -MD -MT tools/python/CMakeFiles/SiloPy.dir/pysilo.cpp.o -MF tools/python/CMakeFiles/SiloPy.dir/pysilo.cpp.o.d -o tools/python/CMakeFiles/SiloPy.dir/pysilo.cpp.o -c /builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.cpp
In file included from /builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/src/silo/silo_private.h:64,
                 from /builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pydbfile.h:55,
                 from /builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.cpp:53:
/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/redhat-linux-build/include/config.h:79:9: warning: ‘HAVE_INTTYPES_H’ redefined
   79 | #define HAVE_INTTYPES_H
      |         ^~~~~~~~~~~~~~~
In file included from /usr/include/python3.14/pyconfig.h:6,
                 from /usr/include/python3.14/Python.h:14,
                 from /builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.cpp:51:
/usr/include/python3.14/pyconfig-64.h:654:9: note: this is the location of the previous definition
  654 | #define HAVE_INTTYPES_H 1
      |         ^~~~~~~~~~~~~~~
In file included from /builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.cpp:54:
/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.cpp: In function ‘PyObject* PyInit_Silo()’:
/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.h:85:13: error: either all initializer clauses should be designated or none of them should be
   85 |             .m_name = name,                     \
      |             ^
/builddir/build/BUILD/Silo-4.12.0-build/Silo-4.12.0/tools/python/pysilo.cpp:257:5: note: in expansion of macro ‘PY_SILO_MOD_DEF’
  257 |     PY_SILO_MOD_DEF(siloModule, "Silo", &SiloMethods[0]);
      |     ^~~~~~~~~~~~~~~
```



Detail on the build here: https://download.copr.fedorainfracloud.org/results/kathenas/generic-fedora-rawhide-all-with-review-not-production/fedora-44-x86_64/10255189-Silo/builder-live.log.gz

--- a/tools/python/pydbfile.cpp
+++ b/tools/python/pydbfile.cpp
@@ -47,6 +47,7 @@
 // reflect those  of the United  States Government or  Lawrence Livermore
 // National  Security, LLC,  and shall  not  be used  for advertising  or
 // product endorsement purposes.
+#include "stdlib.h"
 
 #include "pydbfile.h"
 #include "pydbtoc.h"
@@ -55,6 +56,8 @@
 #include <string>
 
 using std::string;
+
+#define FREE(M)         if(M){free(M);(M)=NULL;}
 
 // ****************************************************************************
 //  Method:  DBfile_DBGetToc
@@ -156,7 +159,7 @@ static PyObject *DBfile_DBGetVar(PyObject *self, PyObject *args)
     int len = DBGetVarLength(db,str);
     if (len < 0)
     {
-        sprintf(&msg[13], "DBGetVarLength");
+        snprintf(&msg[13], sizeof(msg)-13, "DBGetVarLength");
         if (!dontErrInSanityChecks)
             SiloErrorFunc(msg);
         return NULL;
@@ -164,7 +167,7 @@ static PyObject *DBfile_DBGetVar(PyObject *self, PyObject *args)
     int type = DBGetVarType(db,str);
     if (type < 0)
     {
-        sprintf(&msg[13], "DBGetVarType");
+        snprintf(&msg[13], sizeof(msg)-13, "DBGetVarType");
         if (!dontErrInSanityChecks)
             SiloErrorFunc(msg);
         return NULL;
@@ -172,7 +175,7 @@ static PyObject *DBfile_DBGetVar(PyObject *self, PyObject *args)
     void *var = DBGetVar(db,str);
     if (!var)
     {
-        sprintf(&msg[13], "DBGetVar");
+        snprintf(&msg[13], sizeof(msg)-13, "DBGetVar");
         if (!dontErrInSanityChecks)
             SiloErrorFunc(msg);
         return NULL;
@@ -1164,13 +1167,14 @@ static void DBfile_dealloc(PyObject *self)
 //  Creation:    July 12, 2005
 //
 // ****************************************************************************
+#define FSTRSIZE 1000
 static void DBfile_as_string(PyObject *self, char *s)
 {
     DBfileObject *obj = (DBfileObject*)self;
     if (obj->db)
-        sprintf(s, "<DBfile object, filename='%s'>", obj->db->pub.name);
+        snprintf(s, FSTRSIZE, "<DBfile object, filename='%s'>", obj->db->pub.name);
     else
-        sprintf(s, "<closed DBfile object>");
+        snprintf(s, FSTRSIZE, "<closed DBfile object>");
 }
 
 // ****************************************************************************
@@ -1188,7 +1192,7 @@ static void DBfile_as_string(PyObject *self, char *s)
 // ****************************************************************************
 static PyObject *DBfile_str(PyObject *self)
 {
-    char str[1000];
+    char str[FSTRSIZE];
     DBfile_as_string(self, str);
     return PyString_FromString(str);
 }
@@ -1209,7 +1213,7 @@ static PyObject *DBfile_str(PyObject *self)
 // ****************************************************************************
 static int DBfile_print(PyObject *self, FILE *fp, int flags)
 {
-    char str[1000];
+    char str[FSTRSIZE];
     DBfile_as_string(self, str);
     fprintf(fp, "%s\n", str);
     return 0;

--- a/tools/python/pydbfile.h
+++ b/tools/python/pydbfile.h
@@ -52,7 +52,7 @@
 #define PY_DBFILE_H
 
 #include <Python.h>
-#include <silo_private.h>
+#include <silo.h>
 
 // ****************************************************************************
 //  Struct:  DBfileObject

--- a/tools/python/pysilo.h
+++ b/tools/python/pysilo.h
@@ -81,7 +81,7 @@ void SiloErrorFunc(const char *errString);
 #if PY_VERSION_GE(3,0,0)
     #define PY_SILO_MOD_DEF(ob, name, methods)  \
         static struct PyModuleDef moduledef = { \
-            PyModuleDef_HEAD_INIT,              \
+            .m_base = PyModuleDef_HEAD_INIT,    \
             .m_name = name,                     \
             .m_methods = methods };             \
         ob = PyModule_Create(&moduledef);

